src/pyTypeFactory.cc

```diff
@@ -75,6 +75,7 @@ PyType *pyTypeFactory(JSContext *cx, JS::HandleValue rval) {
       if (js::GetProxyHandler(obj)->family() == &PyObjectProxyHandler::family) { // this is one of our proxies for python objects
         return new PyType(JS::GetMaybePtrFromReservedSlot<PyObject>(obj, PyObjectSlot));
       }
+      // TODO PyIteratorProxyHandler
```

Comment:
Missing case here

Reply:
A possible test for this would be:
```py
def test_iter_reen trance():
  myit = iter((1,2))
  result = pm.eval("(iter) => iter")(myit)
  assert myit is result
```

Reply:
Philippe Laporte
  3:41 PM
Thanks. I see little value in supporting this use case and not just a little of work
3:41
I'd be satisfied with this use case working:
3:43
myit = iter((1, 2))
result = [None]
pm.eval("(result, arr) => {result[0] = arr}")(result, myit)
result[0].next()
{'done': False, 'value': 1.0}
3:43
and leave the more advanced support until someone actually uses it
3:44
as a TODO

Caleb
  3:45 PM
would
if (js::GetProxyHandler(obj)->family() == &PyObjectProxyHandler::family) {
        return new PyType(JS::GetMaybePtrFromReservedSlot<PyObject>(obj, PyObjectSlot));
      }
not work, is it more involved than that?

Philippe Laporte
  3:45 PM
yes, that does not work, it is more involved indeed

Caleb
  3:45 PM
then fair enough

Reply:
done